Campus-iOS/Map/MainView/PanelContent.swift

```diff
@@ -49,56 +88,46 @@ struct PanelContent: View {
+                              searchString: $searchString)
                     
                     Spacer().frame(width: 0.25 * UIScreen.main.bounds.width/10,
                                    height: 1.5 * UIScreen.main.bounds.width/10)
                 }
-                ZStack {
-                    ProgressView()
-                    ScrollViewReader { proxy in
-                        List {
-                            ForEach (canteens.indices.filter({ searchString.isEmpty ? true : canteens[$0].name.localizedCaseInsensitiveContains(searchString) }), id: \.self) { id in
-                                PanelRow(cafeteria: self.$canteens[id])
-                                .onTapGesture {
-                                    withAnimation {
-                                        selectedCanteenName = canteens[id].name
-                                        selectedAnnotationIndex = canteens.firstIndex(of: canteens[id])!
-                                        proxy.scrollTo(canteens[id], anchor: .top)
-                                        
-                                        selectedCanteen = canteens[id]
-                                    }
-                                }
-                                .task(id: selectedAnnotationIndex) {
-                                    withAnimation(Animation.linear(duration: 0.0001)) {
-                                        if 0...canteens.count ~= selectedAnnotationIndex {
-                                            proxy.scrollTo(canteens[selectedAnnotationIndex], anchor: .top)
-                                            selectedAnnotationIndex = -1
-                                        }
-                                    }
-                                }
-                            }
-
-                        }
+                
+                List {
+                    ForEach (sortedCanteens.indices.filter({ searchString.isEmpty ? true : sortedCanteens[$0].name.localizedCaseInsensitiveContains(searchString) }), id: \.self) { id in
+                        Button(action: {
+                            selectedCanteen = sortedCanteens[id]
+                        }, label: {
+                            PanelRow(cafeteria: self.$sortedCanteens[id])
+                        })
                     }
-                    .searchable(text: $searchString, prompt: "Look for something")
-                    .listStyle(PlainListStyle())
                 }
-                .onChange(of: selectedCanteenName) { newValue in
-                    print("SELECTED CANTEEN (Panelcontent): ", newValue)
+                .searchable(text: $searchString, prompt: "Look for something")
+                .listStyle(PlainListStyle())
+            }
+        }
+        .onChange(of: selectedCanteen) { optionalCafeteria in
+            if let cafeteria = optionalCafeteria {
+                mealPlanViewModel = MealPlanViewModel(cafeteria: cafeteria)
```

Comment:
As mentioned before, I guess there are cleaner ways to do that. Always instantiating a completely new ViewModel and then making the request upon the creation is a little overkill

Reply:
What could be a cleaner way? 
I felt that reusing the view model and only updating it is even worse, but I didn't know how to do it in another way.

Reply:
Well, just update the data in the viewmodel right? No need to allocate a completly new object.